### PR TITLE
Add: pagination for AllTest (#387)

### PR DESCRIPTION
**Pull request by shivam-sharma7**
## Changes proposed

Add: pagination for AllTest

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by shivam-sharma7** [comment]
> @shivam-sharma7 the backend is failing to compile.
> 
> The REST interface has been changed to return a `TestQueryResult` object:
> 
> https://github.com/Hyperfoil/Horreum/blob/924255e24c7508f06d323ebd0caadbe8b9791a93/horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/services/TestService.java#L47
> 
> However, the implementation is returning a `List<Test>`: https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/TestServiceImpl.java#LL237C11-L237C21
> 
> The `TestServiceImpl` method `public List<Test> list` needs to return `TestQueryResult`, e.g. `public TestQueryResult list` please see what changes were made for the `Schema` patch, and make similar changes for `Test`: [1f8eee3#diff-102ceb75ecfbaf74b880bdbb4fbd034bcb176472b8c17f2dbcdbb281c5237c3a](https://github.com/Hyperfoil/Horreum/commit/1f8eee3c7017d8c38b9ffcbb7947983b7d894cf5#diff-102ceb75ecfbaf74b880bdbb4fbd034bcb176472b8c17f2dbcdbb281c5237c3a)

@johnaohara has done

File: horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/TestServiceImpl.java [hunk 1]
```diff
@@ -257,7 +257,7 @@ public List<Test> list(String roles, Integer limit, Integer page, String sort, S
       if (limit != null && page != null) {
          query.page(Page.of(page, limit));
       }
-      return query.list().stream().map(TestMapper::from).collect(Collectors.toList());
+      return new TestQueryResult query.list().stream().map(TestMapper::from).collect(Collectors.toList()). Test.count();
```
Comment (johnaohara) [inline horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/TestServiceImpl.java:260]: This does not compile, needs to be 

```suggestion
      return new TestQueryResult( query.list().stream().map(TestMapper::from).collect(Collectors.toList()), Test.count() ) ;
```

File: webapp/src/domain/tests/AllTests.tsx [hunk 2]
```diff
@@ -282,6 +286,27 @@ export default function AllTests() {
...
+    const pagination = useMemo(() => ({ page, perPage, direction }), [page, perPage, direction])
+    const [tests, setTets] = useState<TestQueryResult>()
+    const [loading, setLoading] = useState(false)
+
+    useEffect(() => {
+        setLoading(true)
+        Api.testServiceList(
+             SortDirection.Ascending,
+             pagination.perPage,
+             pagination.page
```
Comment (johnaohara) [inline webapp/src/domain/tests/AllTests.tsx:302]: this causes an off-by-one error as the REST API is zero-based and the JS library is one-based,

see: https://github.com/Hyperfoil/Horreum/pull/532/files#diff-4529692787927fec91b2082b9832b678a10586615349fa8663cc0026dcd63c65R38

**Review by johnaohara — CHANGES_REQUESTED** [review]

File: webapp/src/domain/tests/AllTests.tsx (hunk 2 above)
Comment (shivam-sharma7) [reply webapp/src/domain/tests/AllTests.tsx]: Thanks!

**Comment by shivam-sharma7** [comment]
@johnaohara getting compilation error again and again

**Pull request closed without merging**